bluebrain/repo-patches/packages/neuron/package.py

```diff
@@ -1,488 +1,177 @@
-                + ",".join(model for model in self.spec.variants["model_tests"].value)
+                self.define(
+                    "NRN_ENABLE_MODEL_TESTS",
+                    ",".join(model for model in spec.variants["model_tests"].value),
+                )
             )
-        if self.spec.variants["sanitizers"].value != ("None",):
+
+        # sanitizers setup during development
+        if spec.variants["sanitizers"].value != ("None",):
             if self.compiler.name == "clang":
                 args.append(
-                    "-DLLVM_SYMBOLIZER_PATH="
-                    + os.path.join(os.path.dirname(self.compiler.cxx), "llvm-symbolizer")
+                    self.define(
+                        "LLVM_SYMBOLIZER_PATH",
+                        os.path.join(os.path.dirname(self.compiler.cxx), "llvm-symbolizer"),
+                    )
                 )
-            args.append("-DNRN_SANITIZERS=" + ",".join(self.spec.variants["sanitizers"].value))
-        if "+mpi" in self.spec:
-            args.append("-DNRN_ENABLE_MPI=ON")
-            if "+gpu" in self.spec:
-                args.append("-DNRN_ENABLE_MPI_DYNAMIC=OFF")
-            else:
-                args.append("-DNRN_ENABLE_MPI_DYNAMIC=ON")
-        else:
-            args.append("-DNRN_ENABLE_MPI=OFF")
-        if "+python" in self.spec:
-            args.append("-DPYTHON_EXECUTABLE:FILEPATH=" + self.spec["python"].command.path)
-        if self.spec.variants["build_type"].value == "FastDebug":
-            # Do *not* add -DNDEBUG, so assertions are enabled
-            # Good debug information and stack traces
-            compilation_flags.append("-g")
-            if "%nvhpc" not in self.spec:
-                compilation_flags.append("-fno-omit-frame-pointer")
-            # Moderate optimisation by default
-            compilation_flags.append("-O1")
+            args.append(self.define("NRN_SANITIZERS", ",".join(spec.variants["sanitizers"].value)))
+
+        # Before 9.0.a15 CMake didn't have  build_type=FastDebug
+        # If we want to launch BBP CI with such older versions then
+        # we should handle FastDebug build type. At least for some
+        # time until we are sure that we will be using >=9.0.a15 with
+        # FastDebug build type
+        if spec.satisfies("@:9.0.a14") and spec.variants["build_type"].value == "FastDebug":
+            # moderate optimisation by default
+            compilation_flags = ["-g", "-O1"]
             if "%intel" in self.spec:
-                # This one definitely seems wise
+                # this one definitely seems wise
                 compilation_flags += ["-fp-model", "consistent"]
             elif "%oneapi" in self.spec:
-                # The documentation doesn't mention consistent for these intel compilers
+                # the documentation doesn't mention consistent for these intel compilers
                 compilation_flags.append("-fp-model=precise")
```

Comment:
We can't handle floating point handling in CMake?

Reply:
note that this is for `@:9.0.a14` and the CMake change to handle such flags was added only recently. In neurodamus we often need to install older neuron versions for result cross-checking and hence I thought it would be helpful to have this.